src/JSInterop/Microsoft.JSInterop/test/DotNetDispatcherTest.cs

```diff
@@ -390,6 +486,163 @@ public void CannotInvokeWithIncorrectNumberOfParams()
+
+            // Act
+            var result = DotNetDispatcher.ParseArguments("SomeMethod", arguments, new[] { typeof(TestDTO), });
+
+            // Assert
+            var value = Assert.IsType<TestDTO>(Assert.Single(result));
+            Assert.Equal(7, value.IntVal);
+            Assert.Null(value.StringVal);
+        }
+
+        [Fact]
+        public void ParseArguments_NullArgument()
+        {
+            // Arrange
+            var arguments = "[4, null]";
+
+            // Act
+            var result = DotNetDispatcher.ParseArguments("SomeMethod", arguments, new[] { typeof(int), typeof(TestDTO), });
+
+            // Assert
+            Assert.Collection(
+                result,
+                v => Assert.Equal(4, v),
+                v => Assert.Null(v));
+        }
+
+        [Fact]
+        public void ParseArguments_Throws_WithIncorrectDotNetObjectRefUsage()
+        {
+            // Arrange
+            var method = "SomeMethod";
+            var arguments = "[4, {\"__dotNetObject\": 7}]";
+
+            // Act
+            var ex = Assert.Throws<InvalidOperationException>(() => DotNetDispatcher.ParseArguments(method, arguments, new[] { typeof(int), typeof(TestDTO), }));
+
+            // Assert
+            Assert.Equal($"In call to '{method}', parameter of type '{nameof(TestDTO)}' at index 2 must be declared as type 'DotNetObjectRef<TestDTO>' to receive the incoming value.", ex.Message);
+        }
+
+        [Fact]
+        public void ParseEndInvokeArguments_ThrowsIfJsonIsEmptyString()
+        {
+            Assert.ThrowsAny<JsonException>(() => DotNetDispatcher.ParseEndInvokeArguments(new TestJSRuntime(), ""));
+        }
+
+        [Fact]
+        public void ParseEndInvokeArguments_ThrowsIfJsonIsNotArray()
+        {
+            Assert.ThrowsAny<JsonException>(() => DotNetDispatcher.ParseEndInvokeArguments(new TestJSRuntime(), "{\"key\": \"value\"}"));
+        }
+
+        [Fact]
+        public void ParseEndInvokeArguments_ThrowsIfJsonArrayIsInComplete()
+        {
+            Assert.ThrowsAny<JsonException>(() => DotNetDispatcher.ParseEndInvokeArguments(new TestJSRuntime(), "[7, false"));
+        }
+
+        [Fact]
+        public void ParseEndInvokeArguments_ThrowsIfJsonArrayHasMoreThan3Arguments()
+        {
+            Assert.ThrowsAny<JsonException>(() => DotNetDispatcher.ParseEndInvokeArguments(new TestJSRuntime(), "[7, false, \"Hello\", 5]"));
+        }
+
+        [Fact]
+        public void ParseEndInvokeArguments_Works()
+        {
+            var jsRuntime = new TestJSRuntime();
+            var task = jsRuntime.InvokeAsync<TestDTO>("somemethod");
+
+            DotNetDispatcher.ParseEndInvokeArguments(jsRuntime, $"[{jsRuntime.LastInvocationAsyncHandle}, true, {{\"intVal\": 7}}]");
+
+            Assert.True(task.IsCompletedSuccessfully);
+            Assert.Equal(7, task.Result.IntVal);
+        }
+
+        [Fact]
+        public void ParseEndInvokeArguments_UsesStackForSmallPayloads()
```

Comment:
Same question, what does this mean?

Is it a remnant from wanting to test the stackalloc stuff?

Reply:
Yup. Removed in the update